keyboards/handwired/maverick0197/keydeck8/keymaps/default/keymap.c

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2023 Maverick0197 <>
+  *
+  * This program is free software: you can redistribute it and/or modify
+  * it under the terms of the GNU General Public License as published by
+  * the Free Software Foundation, either version 2 of the License, or
+  * (at your option) any later version.
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	 /*
+     * ┌────┬────┬────┐
+     * │ D0 │ D1 │ D4 │
+     * ├────┼────┼────┤
+     * │ C6 │ D7 │ E6 │
+     * ├────┼────┼────┤
+     * │ B4 │ B5 │ B6 │
+     * └────┴────┴────┘
+     */
+	[0] = LAYOUT(
+	 KC_MEDIA_PREV_TRACK, 	KC_MEDIA_PLAY_PAUSE, 	KC_MEDIA_NEXT_TRACK,
+	 KC_1, 	KC_2, 	KC_3,
+	 KC_4, 	KC_5, 	KC_6
+	)
+};
+#if defined(ENCODER_MAP_ENABLE)
+	const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+		[0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+	};
+#else
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    //if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code_delay(KC_VOLU, 10);
+        } else {
+            tap_code_delay(KC_VOLD, 10);
+        }
+    //}
+    return false;
+};
+#endif // ENCODER_MAP_ENABLE

```

Comment:
Ideally, just encoder map is used here. Make sure to create a rules.mk file here to enable it.
```suggestion
#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
        [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
};
#endif
```

Reply:
I tried to use encoder map but I could not get it to work on my device. I will have a second one to test with tomorrow so I will give it another go.
Thanks for the comments by the way, this is my first attempt at this and I have not been able to figure out the "formatting" error I keep getting from lint on here.